website/src/pages/organism-selector/[redirectTo].astro

```diff
@@ -1,31 +1,34 @@
 ---
 import { NeedAGroup } from '../../components/common/NeedAGroup.tsx';
 import NeedToLogin from '../../components/common/NeedToLogin.astro';
-import { getConfiguredOrganisms } from '../../config';
+import { getConfiguredOrganisms, getWebsiteConfig } from '../../config';
 import BaseLayout from '../../layouts/BaseLayout.astro';
 import { routes } from '../../routes/routes';
 import { GroupManagementClient } from '../../services/groupManagementClient';
 import { getAccessToken } from '../../utils/getAccessToken';
 
-const redirectTo = Astro.params.redirectTo;
-const accessToken = getAccessToken(Astro.locals.session)!;
-const groupsResult = await GroupManagementClient.create().getGroupsOfUser(accessToken);
+type ValidRedirectTo = 'submission' | 'search';
 
-const purposes: { [key: string]: string } = {
-    submission: 'to access the submission portal',
+const purposes: Partial<Record<ValidRedirectTo, string>> = {
     search: 'to browse data',
 };
+if (getWebsiteConfig().enableSubmissionPages) {
+    purposes.submission = 'to access the submission portal';
+}
 
-interface Routes {
-    [key: string]: (organism: string) => string;
+if (!(Astro.params.redirectTo! in purposes)) {
+    return Astro.rewrite('/404');
 }
```

Comment:
Why do we need to redirect here? Wouldn't it be catched by the submissionPagesDisablingMiddleware?

Reply:
The middleware currently doesn't catch this page (as is separate from `[organism]/submission`). We could include this in the middleware but I found it cleaner to do it directly here. (No strong preference though)